common/djangoapps/track/eventshim.py

```diff
@@ -0,0 +1,380 @@
+"""
+EventShims that create
+"""
+
+# pylint: disable=missing-docstring
+
+import json
+import logging
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import UsageKey
+
+log = logging.getLogger(__name__)
+
+
+class DottedPathMapping(object):
+    """
+    Dictionary-like object for creating keys of dotted paths.
+
+    If a key is created that ends with a dot, it will be treated as a path
+    prefix.  Any value whose prefix matches the dotted path can be used
+    as a key for that value, but only the most specific match will
+    be used.
+    """
+
+    def __init__(self, registry=None):
+        self._match_registry = {}
+        self._prefix_registry = {}
```

Comment:
Why was it necessary to distinguish between prefixes ending with a dot, versus exact matches?  Why not just have a dictionary where the keys are regexes?